Description: use iconv instead of iconv-lite
 Before, body-parser depended on iconv-lite which has a slightly different
 API. Now body-parser depends on iconv.
Author: Andrew Kelley <superjoe30@gmail.com>
Forwarded: not-needed
Last-Update: 2014-06-30

--- node-body-parser-1.4.3.orig/lib/read.js
+++ node-body-parser-1.4.3/lib/read.js
@@ -8,7 +8,7 @@
  */
 
 var getBody = require('raw-body')
-var iconv = require('iconv-lite')
+var Iconv = require('iconv').Iconv;
 var typer = require('media-typer')
 var zlib = require('zlib')
 
@@ -91,7 +91,7 @@ function read(req, res, next, parse, opt
     // parse
     try {
       body = typeof body !== 'string' && encoding !== null
-        ? iconv.decode(body, encoding)
+        ? iconvDecode(body, encoding)
         : body
       req.body = parse(body)
     } catch (err){
@@ -111,6 +111,11 @@ function read(req, res, next, parse, opt
   }
 }
 
+function iconvDecode(body, encoding) {
+  var iconv = new Iconv(encoding, 'utf-8');
+  return iconv.convert(body).toString('utf8');
+}
+
 /**
  * Get the content stream of the request.
  *
